[DOC] Add documentation for `StrimziPodSets` and `UseStrimziPodSet` feature gate (#6312)
strimzi/strimzi-kafka-operator: documentation/modules/operators/ref-operator-cluster.adoc

NOTE: The `ServiceAccountPatching` feature gate moved to beta stage in Strimzi 0.27.0 and is expected to remain in the beta stage until Strimzi 0.30.

[id='ref-operator-use-strimzi-pod-sets-feature-gate-{context}']
=== Use `StrimziPodSets` feature gate

> Currently, Strimzi relies on StatefulSets to create and manage Pods for the ZooKeeper and Kafka clusters.

Reviewer: If we're not referring directly to the object/resource with backticks (`Pod`) we can use lower case like pods, nodes etc. StatefulSets could be split to be _stateful sets_.

Strimzi creates the StatefulSet and Kubernetes creates the Pods according to the StatefulSet definition.
When a Pod is deleted, Kubernetes is responsible for recreating it.

> The use of StatefulSets is also a cause of several limitations.
> For example:

Reviewer: ```suggestion
The use of StatefulSets has the following limitations:
```

> * The Pods are always created / removed based on their index numbers

Reviewer: ```suggestion
* Pods are always created or removed based on their index numbers
```

* All Pods in the StatefulSet need to have a similar configuration
* Changing storage configuration for the Pods in the StatefulSet is complicated

The `UseStrimziPodSets` feature gate introduces our own resource for managing the pods called `StrimziPodSet`.

> When the feature gate is enabled, this resource is used instead of the StatefulSets and Strimzi handles the creation and management of Pods itself.

Reviewer: ```suggestion
When the feature gate is enabled, this resource is used instead of the StatefulSets. Strimzi handles the creation and management of pods instead of Kubernetes.
```

> Using StrimziPodSets instead of StatefulSets should allow us to add more new features in the future.

Reviewer: ```suggestion
Using StrimziPodSets instead of StatefulSets provides more control over the functionality.
```
